fix: normalise all Chainlink price feed answers to 18 decimals

lumoswiz opened the pull request:
This PR addresses the potential issue raised in our recent discussion where the `SignedWadMath` functions like `wadMul` and `wadPow` expect all arguments to be in WAD format (1e18 precision). 

This PR is in response to an issue raised [here](https://github.com/cowdao-grants/cow-amm-lp-oracle/pull/49#discussion_r1967987526).

### Changes
- Remove conditional that only normalised feed answers when feed decimals differed
- Always normalize Chainlink feed answers to 18 decimals regardless of their original decimal basis
- Adjust test suite to expect 18 decimal normalised values

### Impact
This change ensures that all internal calculations using `wadMul`, `wadPow`, and other WAD-based math functions will work correctly with properly scaled values. 

For integration testing with Aave V3, we've updated our AaveLPOracle adapter to convert from our new 18 decimal standard back to Aave's expected 8 decimal format.

lumoswiz committed:
fix: normalise all chainlink feed answers to 18 decimals

Remove conditional that only normalized when feed decimals differ

lumoswiz committed:
fix: synchronise naive price calculations with normalised chainlink feed answers

lumoswiz committed:
fix: revise LPOracle fork test assertions to use relative comparison for price stability

lumoswiz committed:
fix: normalize Aave mock oracle output to 8 decimals in integration tests

lumoswiz committed:
fix: adjust test calculations to convert 18-decimal answers to Aave's 8-decimal standard

lumoswiz committed:
test: update getFeedData unit tests to expect 18-decimal normalised values

lumoswiz committed:
fix: adjust feed data fuzz tests to expect 18-decimal normalised values

lumoswiz committed:
fix: update LatestRoundData tests to use 18-decimal normalised values

lumoswiz committed:
fix: warn on max line length

bh2smith reviewed (approved):
Great stuff. Looks good. I like how the Aave integration was updated.

fleupold commented on src/LPOracle.sol:
```diff
@@ -128,7 +128,7 @@ contract LPOracle is AggregatorV3Interface {
     |*----------------------------------------------------------*/
 
-    /// @notice Retrieves latest price data from Chainlink feeds and adjusts for decimals.
+    /// @notice Retrieves latest price data from Chainlink feeds and normalise answers to 18 decimals.
```
Doesn't the `decimals()` function also need to be updated? I think it might be a nice addition to the unit test to also assert what `decimals()` returns next to when we assertApproxEq on the result.

fleupold commented on test/integration/Integration.t.sol:
```diff
@@ -246,7 +249,7 @@ contract IntegrationTest is Addresses, BaseTest {
         // Don't repay, gain: totalDebtBase - totalCollateralBase
         // Approx. 103k in this scenario
-        assertGt(totalDebtBase, totalCollateralBase);
+        assertGt(totalDebtBase, totalCollateralBase, "this one");
```
surprising change

lumoswiz replied:
Will remove. Slipped through.

bh2smith replied:
[dbb9fcd](https://github.com/cowdao-grants/cow-amm-lp-oracle/pull/54/commits/dbb9fcd05bf3987e994710e0d1e9d719435e4a5f)

lumoswiz committed:
fix: remove debug message from pool manipulation test assertion

lumoswiz committed:
refactor: standardise decimals() to return 18

lumoswiz committed:
test: simplify decimals fork test to expect fixed 18 decimals

lumoswiz committed:
test: add decimals() verification and standardise price notation

Add decimals() assertions to unit tests
Use explicit 10 ** 18 format for price expectations

bh2smith reviewed (approved):
Parfait!
